ibis/tests/all/test_string.py

```diff
@@ -150,6 +150,33 @@ def test_string_col_is_unicode(backend, alltypes, df):
             lambda t: t.date_string_col.str[-2:],
             id='right',
         ),
+        param(
+            lambda t: t.date_string_col[1:3],
+            lambda t: t.date_string_col.str[1:3],
+            id='slice'
+        ),
+        param(
+            lambda t: t.date_string_col[t.date_string_col.length() - 1:],
```

Comment:
In a follow-up PR we could add support for translating negative indices to `col.length() + <negative index>`

Reply:
yes, though why cross out the follow up?

Reply:
i don't think that should happen in this PR